### PR TITLE
Make MedCon and Tag comparable

### LINKED ISSUE #96: Streamline code that require sets initialisation
lots of repeated code so add a utility
function to streamline it and ensure it
works for all related classes

### PR DESCRIPTION
**Pull request by TechRaj**
Fixes #96

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddMedConCommand.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class AddMedConCommand extends Command {
 
     /**
      * @param nric of the patient to assign the medical conditions to
-     * @param medCons set of medical conditions to be assigned
+     * @param medCons set of medical conditions to be assigned.
```
Comment (yadobler) [inline src/main/java/seedu/address/logic/commands/AddMedConCommand.java:45]: good catch

File: src/main/java/seedu/address/model/person/MedCon.java [hunk 2]
```diff
@@ -46,4 +46,9 @@ public boolean equals(Object other) {
     public int hashCode() {
         return value.hashCode();
     }
+
+    @Override
+    public int compareTo(MedCon other) {
+        return this.value.compareTo(other.value);
+    }
```
Comment (yadobler) [inline src/main/java/seedu/address/model/person/MedCon.java:53]: maybe just add in the header javadoc that this class's natural ordering is alphabetical order of value

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 3]
```diff
@@ -59,4 +59,8 @@ public String toString() {
         return '[' + tagName + ']';
     }
 
+    @Override
+    public int compareTo(Tag other) {
+        return this.tagName.compareTo(other.tagName);
+    }
```
Comment (yadobler) [inline src/main/java/seedu/address/model/tag/Tag.java:65]: same here, add in the header javadoc that this class's natural ordering is alphabetical order of value

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 4]
```diff
@@ -85,19 +86,29 @@ public JsonAdaptedPerson(Person source) {
...
         nric = source.getNric().value;
-        medCons.addAll(source.getMedCons()
-                .stream()
-                .map(JsonAdaptedMedCon::new)
-                .toList());
+
+        // Use helper function to convert and sort medical conditions, tags, and appointments
+        medCons.addAll(convertToSortedList(source.getMedCons(), JsonAdaptedMedCon::new));
+        tags.addAll(convertToSortedList(source.getTags(), JsonAdaptedTag::new));
+        appointments.addAll(convertToSortedList(source.getAppointments(), JsonAdaptedAppointment::new));
```
Comment (yadobler) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:93]: beautiful. SLAPs hard

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 5]
```diff
@@ -85,19 +86,29 @@ public JsonAdaptedPerson(Person source) {
...
+     * @param mapper A function to map the item to its adapted form.
+     * @param <T> The type of items in the set.
+     * @param <R> The type of the adapted JSON item.
+     * @return A sorted list of adapted items.
+     */
+    private <T extends Comparable<T>, R> List<R> convertToSortedList(Set<T> source, Function<T, R> mapper) {
+        return source.stream()
+                .sorted()
+                .map(mapper)
+                .toList();
```
Comment (yadobler) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:111]: beautiful

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 6]
```diff
@@ -69,16 +68,16 @@ public PersonCard(Person person, int displayedIndex) {
...
+        addLabelsToFlowPane(person.getMedCons(), medCons);
+        addLabelsToFlowPane(person.getTags(), tags);
+        addLabelsToFlowPane(person.getAppointments(), appointments);
     }
+
+    private <T extends Comparable<T>> void addLabelsToFlowPane(Set<T> items, FlowPane flowPane) {
+        items.stream()
+                .sorted(Comparator.naturalOrder())
+                .forEach(item -> flowPane.getChildren().add(new Label(item.toString())));
+    }
```
Comment (yadobler) [inline src/main/java/seedu/address/ui/PersonCard.java:81]: have you smoke-tested it? does everything show up when running the actual prog?

**Review by yadobler — COMMENTED** [review]
LGTM

if the questions i posted are resolved then it's a great addition

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 6 above)
Comment (TechRaj) [reply src/main/java/seedu/address/ui/PersonCard.java:81]: Yep the sample data loads up

**Review by yong-zhuo — APPROVED** [review]
LGTM

**Review by PresleyChew — COMMENTED** [review]
Overall LGTM, neat!

**Pull request merged**